### ISSUE TITLE
RDS does not support creating a DB instance with the following combination

### ISSUE THREAD

**Issue by BwL1289** [aws-cloudformation/cfn-lint#4048]
### CloudFormation Lint Version

cfn-lint 1.28.0

### What operating system are you using?

MacOS

### Describe the bug

I get the following error in cloudformation:

```
Resource handler returned message: "RDS does not support creating a DB instance with the following combination: DBInstanceClass=db.t2.small, Engine=mysql, EngineVersion=8.0.39, LicenseModel=general-public-license. For supported combinations of instance class and database engine version, see the documentation. (Service: Rds, Status Code: 400, Request ID: b3d19f56-de1f-4e0d-bb0d-e3542107d159) (SDK Attempt Count: 1)" (RequestToken: 94df6532-b0ab-64e9-e748-29e5f0cc0081, HandlerErrorCode: InvalidRequest)
```

with the configuration below. It does not raise an error in cfnlint when it should.

### Expected behavior

Should raise an error for all incompatible configurations in RDS.

### Reproduction template

```yaml
RdsSvcDatabaseInstance136469D8:
    Type: AWS::RDS::DBInstance
    Properties:
      AllocatedStorage: "100"
      AutoMinorVersionUpgrade: true
      CopyTagsToSnapshot: true
      DBInstanceClass: db.t2.small
      DBParameterGroupName:
        Ref: RdsMySqlParameterGroup5F09708E
      DBSubnetGroupName:
        Ref: MySqlDatabaseSubnetGroup94ACF365
      Engine: mysql
      EngineVersion: 8.0.39
      MasterUserPassword:
        <redacted>
      MasterUsername:
        <redacted>
      OptionGroupName: default:mysql-8-0
      PubliclyAccessible: true
      StorageType: standard
      VPCSecurityGroups:
        - Fn::GetAtt:
            - RdsSecurityGroup5F48F67B
            - GroupId
    UpdateReplacePolicy: Delete
    DeletionPolicy: Delete
```

**Comment by kddejong** [comment]
Looks like we could build this combination using something like 

`aws rds describe-orderable-db-instance-options --engine aurora-postgresql --query "*[].{DBInstanceClass:DBInstanceClass,StorageType:StorageType}|[?StorageType=='gp2']|[].{DBInstanceClass:DBInstanceClass}" --output text --region "us-east-1"`

Not sure how much data is here and how hard it would be to put into some schemas.

**Comment by BwL1289** [comment]
There's likely a good amount of data in there. Without Cfnlint raising errors here though, it's frustrating to get the correct combination, so would love to see this included
